### PR TITLE
Fixes generate static variable and generate static function

### PR DESCRIPTION
**Pull request by SlavaRa**
BUG in release 5.1.1, code for example:
```
package {
	public class Main {
		public function Main() {
			Main.test|;
		}
	}
}
```
result after generating variable:
```
package {
	public class Main {
		private var test;//must be private static var test;
		public function Main() {
			Main.test;
		}
	}
}
```

BUG in developmnet version if setting generate explicit scope is true, code for example:
```
package;
public class Main {
	public static function main() {
		test|;
	}
}
```
result after generating variable:
```
package;
public class Main {
	static private var test;
	public static function main() {
		this.test;//must be Main.test
	}
}
```

### CODE REVIEW

**Comment by SlavaRa** [comment]
@Neverbirth please review this PR

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 1]
```diff
@@ -2081,13 +2081,15 @@ private static void GenerateVariableJob(GeneratorJobType job, ScintillaControl s
...
+            var memberIsStatic = member != null && (member.Flags & FlagType.Static) > 0;
             if (ASContext.CommonSettings.GenerateScope && !varResult.Context.Value.Contains(ASContext.Context.Features.dot))
             {
                 position = sci.CurrentPos;
                 var start = sci.WordStartPosition(position, true);
                 var length = sci.MBSafeTextLength(contextToken);
                 sci.SetSel(start, start + length);
-                var text = "this." + contextToken;
+                var scope = memberIsStatic ? inClass.QualifiedName : "this";
+                var text = string.Format("{0}.{1}", scope, contextToken);
```
Comment (wise0704) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: I suggest never using `string.Format()` in completion/parser codes. They are extremely slow.
Use `scope + "." + contextToken` or `string.Concat(scope, ".", contextToken)` instead.

Also, wouldn't `ASContext.Context.Features.dot` be better than `"."`?

Comment (Neverbirth) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: Speed is not a concern in this case anyway. About the dot you are right.

Comment (SlavaRa) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: fixed ;)

**Comment by SlavaRa** [comment]
@Neverbirth @elsassph, guys what about this PR?

**Comment by Neverbirth** [comment]
I will try to give a final reply for the PR between today and tomorrow, but so far it's fine I think.

**Comment by Meychi** [comment]
If @Neverbirth accepts its good for merge.

**Comment by Neverbirth** [comment]
Good to go.

**Comment by SlavaRa** [comment]
thanks

**Pull request merged**